### PR TITLE
Add automated tests

### PR DESCRIPTION
**Pull request by mikeconley**
I probably should have found a way of updating https://github.com/amccreight/ipdl_parser/pull/11 instead, but I ended up rebasing some commits, and so I had to blow away my old branch due to re-written history.

So this has 3 commits. The first one moves all of the ok / error IPDL from mozilla-central over to this repo. The second adds a very simple smoke test framework that runs when you call `cargo test`, and the last commit disables a bunch of the files that happened to not pass when I ran the tests. Sometimes that's because we returned None when we should have returned Some. Sometimes it's because we panic. In any case, I hope we can address those in follow-ups.

### CODE REVIEW

File: tests/smoke_test.rs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+const BASE_PATH: [&'static str; 2] = [".", "tests"];
+const OK_PATH: &'static str = "ok";
+const ERROR_PATH: &'static str = "error";
+
+fn test_enabled_files(test_file_path: &str, cb: &Fn(Option<HashMap<PathBuf, TranslationUnit>>)) {
+    let mut path: PathBuf = BASE_PATH.iter().collect();
+    path.push(test_file_path);
+
+    let mut include_dirs = Vec::new();
+    include_dirs.push(path.clone());
```
Comment (amccreight) [inline tests/smoke_test.rs]: The more convenient way to make a singleton Vec is let foo = vec![1, 2, 3 4];

File: tests/smoke_test.rs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    let entries = fs::read_dir(&path).expect("Should have the test file directory");
+    for entry in entries {
+        if let Ok(entry) = entry {
+            if entry.path().extension().unwrap() == "disabled" {
+                println!("Skipping {:?}", entry.file_name());
+            } else {
+                let mut file_names = Vec::new();
```
Comment (amccreight) [inline tests/smoke_test.rs]: You can also use vec! here.

File: tests/smoke_test.rs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+#[test]
+fn error_tests() {
+    fn assert_is_none(tus: Option<HashMap<PathBuf, TranslationUnit>>) {
+        assert!(tus.is_none());
+    }
+
+    let assert_function = assert_is_none;
+    test_enabled_files(ERROR_PATH, &assert_function);
+}
```
Comment (amccreight) [inline tests/smoke_test.rs]: Add a newline here. I guess.

File: tests/smoke_test.rs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+                let tus = ipdl_parser::parser::parse(&include_dirs, file_names);
+                cb(tus);
+            }
+        }
+    }
+}
+
+#[test]
+fn ok_tests() {
+    fn assert_is_some(tus: Option<HashMap<PathBuf, TranslationUnit>>) {
```
Comment (amccreight) [inline tests/smoke_test.rs]: This whole callback thing seems overengineered, but I suppose it is fine.

Comment (mikeconley) [inline tests/smoke_test.rs]: I'm definitely willing to accept feedback. Would you be okay with us just repeating the same routine to iterate the non-disabled files - once in ok_tests and again in error_tests?

Comment (amccreight) [inline tests/smoke_test.rs]: Nah, its fine. I was imagining just passing in a boolean or something instead but it doesn't matter much.

File: tests/smoke_test.rs (hunk 1 above)
Comment (mikeconley) [inline tests/smoke_test.rs]: Ah, very handy - thank you. Did this here and in the other location as well.

File: tests/smoke_test.rs (hunk 3 above)
Comment (mikeconley) [inline tests/smoke_test.rs]: Done!

**Pull request merged**